src/app/diary-module/containers/diary-home/diary-home.component.ts

```diff
@@ -1,25 +1,43 @@
 import { Component, OnInit } from '@angular/core'
 import { FormControl, FormGroup, Validators } from '@angular/forms'
+import { Card } from 'src/app/models/card'
 
 @Component({
     selector: 'app-diary-home',
     templateUrl: './diary-home.component.html',
     styleUrls: ['./diary-home.component.scss'],
 })
 export class DiaryHomeComponent implements OnInit {
-    form = new FormGroup({
-        title: new FormControl('', Validators.required),
-        description: new FormControl('', Validators.required),
-    })
+    form = new FormGroup(
+        {
+            title: new FormControl('', Validators.required),
+            description: new FormControl('', Validators.required),
+        },
+        Validators.required
+    )
     expanded: boolean = false
+    cards: Card[] = []
 
     constructor() {}
 
     ngOnInit(): void {}
 
     handleSubmit() {
-        console.log('Title : ' + this.form.value.title)
-        console.log('Description : ' + this.form.value.description)
+        const validTitle = this.form.controls.title.valid
+        const validDescription = this.form.controls.description.valid
+
+        if (validTitle && validDescription) {
```

Comment:
simple you can use `this.form.valid()`. after that rest of the logic should be changed accordingly